code/modules/integrated_electronics/core/assemblies.dm

```diff
@@ -0,0 +1,379 @@
+#define IC_COMPONENTS_BASE		25
+#define IC_COMPLEXITY_BASE		75
+
+/obj/item/device/electronic_assembly
+	name = "electronic assembly"
+	desc = "It's a case, for building small electronics with."
+	w_class = WEIGHT_CLASS_SMALL
+	icon = 'icons/obj/electronic_assemblies.dmi'
+	icon_state = "setup_small"
+	flags_1 = NOBLUDGEON_1
+	var/max_components = IC_COMPONENTS_BASE
+	var/max_complexity = IC_COMPLEXITY_BASE
+	var/opened = 0
+	var/obj/item/stock_parts/cell/battery = null // Internal cell which most circuits need to work.
+	var/cell_type = /obj/item/stock_parts/cell
+	var/can_charge = 1 //Can it be charged in a recharger?
+	var/charge_sections = 4
+	var/charge_tick = 0
+	var/charge_delay = 4
+	var/use_cyborg_cell = 1
+
+/obj/item/device/electronic_assembly/medium
+	name = "electronic mechanism"
+	icon_state = "setup_medium"
+	desc = "It's a case, for building medium-sized electronics with."
+	w_class = WEIGHT_CLASS_SMALL
+	max_components = IC_COMPONENTS_BASE * 2
+	max_complexity = IC_COMPLEXITY_BASE * 2
+
+/obj/item/device/electronic_assembly/large
+	name = "electronic machine"
+	icon_state = "setup_large"
+	desc = "It's a case, for building large electronics with."
+	w_class = WEIGHT_CLASS_BULKY
+	max_components = IC_COMPONENTS_BASE * 4
+	max_complexity = IC_COMPLEXITY_BASE * 4
+	anchored = 0
+
+/obj/item/device/electronic_assembly/large/attackby(var/obj/item/O as obj, var/mob/user as mob)
+	if(istype(O, /obj/item/wrench))
+		if(!istype(src.loc,/turf))
+			return
+		src.anchored = !anchored
+		to_chat(user,"You [src.anchored ? "wrench" : "unwrench"] \the [src].")
+		return
+	..()
+
+/obj/item/device/electronic_assembly/large/attack_tk(mob/user as mob)
+	if(anchored)
+		return
+	..()
+
+/obj/item/device/electronic_assembly/large/attack_hand(mob/user as mob)
+	if(anchored)
+		src.attack_self(user)
+		return
+	..()
+
+/obj/item/device/electronic_assembly/drone
+	name = "electronic drone"
+	icon_state = "setup_drone"
+	desc = "It's a case, for building mobile electronics with."
+	w_class = WEIGHT_CLASS_SMALL
+	max_components = IC_COMPONENTS_BASE * 3
+	max_complexity = IC_COMPLEXITY_BASE * 3
+
+
+
+/obj/item/device/electronic_assembly/New()
+	..()
```

Comment:
probably   . = ...()

Reply:
It's `. = ..()`

Also, `Initialize()`

Reply:
Yeah phone caused that.

Reply:
No, that's right. Call version of parent, then spawn battery and start processing.All is correct.

Reply:
Still needs to be Initialize